activerecord/test/cases/connection_adapters/schema_cache_test.rb

```diff
@@ -95,6 +95,10 @@ def test_caches_database_version
 
         assert_no_queries do
           assert_equal @database_version.to_s, @cache.database_version.to_s
+
+          if current_adapter?(:Mysql2Adapter)
```

Comment:
I'm really not liking the idea that we're now introducing conditional storage in the schema cache. YAML caching/reading have been pain points in the past. See https://github.com/rails/rails/pull/26017 and https://github.com/rails/rails/commit/6eb978234cdce57e75c36a096fa76a634705c7c8

Reply:
It's not exactly conditional behavior in the schema cache. Each adapter is responsible for defining `get_database_version` and that gets stored in the schema cache. `Mysql2Adapter` happens to add more information in it's representation of the version.